www/writeLoginApi.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+use SimpleSAML\Logger;
+use SimpleSAML\Module\proxystatistics\Config;
+use SimpleSAML\Module\proxystatistics\DatabaseCommand;
+
+if ('POST' !== $_SERVER['REQUEST_METHOD'] && 'PUT' !== $_SERVER['REQUEST_METHOD']) {
+    Logger::info(
+        'proxystatistics:writeLoginApi - API write called not using POST nor PUT, returning 405 response code'
+    );
+    header('HTTP/1.0 405 Method Not Allowed');
+    exit;
+}
+$config = Config::getInstance();
+
+if (!$config->isApiWriteEnabled()) {
+    Logger::info(
+        'proxystatistics:writeLoginApi - API write called, but disabled in config. Returning 501 response code'
+    );
+    header('HTTP/1.0 501 Not Implemented');
+    echo '<p>Access denied. You did not enter a password.</p>';
+    exit;
+}
+
+$authUsername = $_SERVER['PHP_AUTH_USER'] ?? '';
+$authPass = $_SERVER['PHP_AUTH_PW'] ?? '';
+
+$username = $config->getApiWriteUsername();
+$password = $config->getApiWritePassword();
+
+// If we get here, username was provided. Check password.
+if ($authUsername !== $username || $authPass !== $password) {
```

Comment:
```suggestion
if ($authUsername !== $username || !password_verify($authPass, $passwordHash)) {
```
Direct comparison of plaintext password is insecure, it is vulnerable to timing attacks and the password is in plaintext in config.

Store password's hash in config (result of [password_hash](https://www.php.net/manual/en/function.password-hash.php)) and then use password_verify.